tests/transforms/test_qcut.py

```diff
@@ -1582,6 +1722,233 @@ def test_no_measure_node_observables(self):
             compare_tapes(tape, config)
 
 
+class TestExpandFragmentTapesMC:
+    """
+    Tests fragment tapes are correctly expanded to all random configurations
+    for the Monte Carlo sampling technique.
+    """
+
+    def test_expand_mc(self, monkeypatch):
+        """
+        Tests that fragment configurations are generated correctly using the
+        `expand_fragment_tapes_mc` method.
+        """
+        with qml.tape.QuantumTape() as tape0:
+            qml.Hadamard(wires=[0])
+            qml.CNOT(wires=[0, 1])
+            qcut.MeasureNode(wires=[1])
+            qml.sample(qml.Projector([1], wires=[0]))
+
+        with qml.tape.QuantumTape() as tape1:
+            qcut.PrepareNode(wires=[1])
+            qml.CNOT(wires=[1, 2])
+            qml.sample(qml.Projector([1], wires=[1]))
+            qml.sample(qml.Projector([1], wires=[2]))
+
+        tapes = [tape0, tape1]
+
+        edge_data = {"pair": (tape0.operations[2], tape1.operations[0])}
+        communication_graph = MultiDiGraph([(0, 1, edge_data)])
+
+        with monkeypatch.context() as m:
+            m.setattr(
+                np.random, "choice", lambda a, size=(1, 3), replace=True: np.array([[4, 0, 1]])
```

Comment:
```suggestion
                np.random, "choice", lambda a, size, replace: np.array([[4, 0, 1]])
```
Do we need the defaults?

Reply:
manually checked and added here: f9b8ddf4bc34bf64ce9b0d9743331d157abbf9b3

Thanks!